packages/program/src/revert/revert-error.ts

```diff
@@ -62,8 +62,18 @@ export class RevertError extends Error {
    * @param receipt - The transaction revert receipt.
    * @param reason - The revert reason.
    */
-  constructor(receipt: TransactionResultRevertReceipt, reason: RevertReason) {
-    super(`The script reverted with reason ${reason}`);
+  constructor(receipt: TransactionResultRevertReceipt, reason: RevertReason, logs: Array<unknown>) {
+    // @TODO: this is a "hack"
+    // motivation: a custom error revert string is the only "string" type item in the `logs` array
+    // the `logs.filter for string` will make sense only for `RequireFailed` errors, so could potentially
+    // just restrict the following to `RequireFailed` class by modifying `this.message`
+    // produces error messages like so:
+    // ```RequireRevertError: The script reverted with reason RequireFailed (reason: "FailImplicitly")```
```

Comment:
Not a fan of todo's in the codebase, would advise removing and creating an issue to follow up
```suggestion

```

Reply:
I think we can just remove it then. We don't need a follow-up because we already have proper support for the decoding of str slices in the v1 encoding scheme.